### ISSUE TITLE
Same host *.example.com is configured with different VirtualServerAddress

### ISSUE THREAD

**Issue by mitchellmaler** [F5Networks/k8s-bigip-ctlr#2206]
#### Setup Details
 CIS Version : 2.7.1      
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP BIG-IP 15.1.2 Build 0.0.9 Final
 AS3 Version: BIG-IP 15.1.2 Build 0.0.9 Final
 Agent Mode: AS3  
 Orchestration: K8S
 Orchestration Version: v1.21.6
 Pool Mode: Nodeport      

#### Description  
Create two VirtualServers with a different address but sharing the same wildcard domain hostname. 
The controller throws an error "Same host *.example.com is configured with different VirtualServerAddress"

#### Steps To Reproduce
1) Create two VirtualServers with the same Host such as *.example.com but a different IP
2) Create the TLS Profile shared by both using the same *.example.com in the hosts property
3)

#### Expected Result 
Both VirtualServers are created on the device with the same *.example.com policy but different IP.

#### Actual Result  
CIS throws an error processing those VirtualServer resources

#### Diagnostic Information
```buildoutcfg
<Configuration files, error messages, logs>
Note: Sanitize the data. For example, be mindful of IPs, ports, application names and URLs
Note: The following F5 article outlines the information required when opening an issue.
https://support.f5.com/csp/article/K60974137
```

**Comment by trinaths** [comment]
@mitchellmaler please share example CRDs to validate and further investigate this issue.

**Comment by trinaths** [comment]
@mitchellmaler please share example CRDs to validate and further investigate this issue.

**Comment by mitchellmaler** [comment]
Hi @trinaths sorry for the delay. Here are two example VirtualServers that share the same domain but have a different IP. 
On 2.6 we did not have to set the hostnames on the vs or tls policy but found out that was a requirement when upgrading to 2.7. Using these same VirtualServers (and the tls profile they share) I set the host to *.example.com but it complained they cannot be the same.

```yaml
apiVersion: cis.f5.com/v1
kind: VirtualServer
metadata:
  annotations:
    domian: example.com
  labels:
    app.kubernetes.io/managed-by: pulumi
    f5cr: "true"
  name: example-internal-https
  namespace: istio-system
spec:
  httpTraffic: allow
  iRules:
  - /Common/SSL_308_Redirect
  - /Common/X-Forwarded-Proto
  - /Common/X-Request-Start
  - /Common/X-Forwarded-For
  - /Common/Strip-Server-Name
  - /Common/Strict-Transport-Security
  pools:
  - monitor:
      interval: 3
      send: "GET / HTTP/1.1\r\nHost: \r\nConnection: Close\r\n\r\n"
      timeout: 5
      type: http
    path: /
    service: istio-ingressgateway-internal
    servicePort: 443
  tlsProfileName: example-tls-profile
  virtualServerAddress: 10.182.221.51
  virtualServerName: example-internal-https

apiVersion: cis.f5.com/v1
kind: VirtualServer
metadata:
  annotations:
    domian: example.com
  labels:
    app.kubernetes.io/managed-by: pulumi
    f5cr: "true"
  name: example-external-https
  namespace: istio-system
spec:
  httpTraffic: allow
  iRules:
  - /Common/SSL_308_Redirect
  - /Common/X-Forwarded-Proto
  - /Common/X-Request-Start
  - /Common/X-Forwarded-For
  - /Common/Strip-Server-Name
  - /Common/Strict-Transport-Security
  pools:
  - monitor:
      interval: 3
      send: "GET / HTTP/1.1\r\nHost: \r\nConnection: Close\r\n\r\n"
      timeout: 5
      type: http
    path: /
    service: istio-ingressgateway-external
    servicePort: 443
  tlsProfileName: example-tls-profile
  virtualServerAddress: 10.182.221.50
  virtualServerName: example-external-https
```

**Comment by mdditt2000** [comment]
@mitchellmaler do you have a few minutes to discuss some enhancement available in CIS 2.7 

- Policy CRD for your iRules, This will simplify you VS CRD
- Re-use the VirtualServer Public IP with the multi-host flag
- use wildcard for the domain name *.example.com

Thanks Mark

**Comment by mitchellmaler** [comment]
Sure @mdditt2000

**Comment by mitchellmaler** [comment]
@mdditt2000  I am free tomorrow and Friday afternoon CST if that works?

**Comment by mdditt2000** [comment]
@mitchellmaler sent you a email

**Comment by mdditt2000** [comment]
@trinaths recreated issue with @mitchellmaler 
- issue is blocking upgrade. 

Logs

```
2022/02/03 21:47:51 [DEBUG] Enqueueing VirtualServer: &{{ } {vs-tea  default /apis/cis.f5.com/v1/namespaces/default/virtualservers/vs-tea ca2d1484-0734-45cc-a8d7-faa7d3826d40 115151245 1 2022-02-03 21:49:36 +0000 UTC <nil> <nil> map[f5cr:true] map[] [] []  [{kubectl-cr03 21:49:36 +0000 UTC FieldsV1 {"f:metadata":{"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:host":{},"f:pools":{},"f:snat":{},"f:virtualServerAddress":{}}}}]} {*.example.com  10.192.75.118   0 0 [{/ f5-demo 80  {http /  20 10} }]   auto   [] [] [] } { }}
2022/02/03 21:47:51 [DEBUG] Processing Key: &{default VirtualServer vs-tea 0xc0001e4780 false}
2022/02/03 21:47:51 [DEBUG] Process all the Virtual Servers which share same VirtualServerAddress
2022/02/03 21:47:51 [ERROR] Same host *.example.com is configured with different VirtualServerAddress :
2022/02/03 21:47:51 [DEBUG] Finished syncing virtual servers &{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:vs-tea GenerateName: Namespace:default SelfLink:/apis/cis.f5.com/v1/namespaces/default/virtualservers/vs-tea UID:ca2d1484-0734-45cc-a8d7-faa7d3826d40 ResourceVeionTimestamp:2022-02-03 21:49:36 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[f5cr:true] Annotations:map[] OwnerReferences:[] Finalizers:[] ClusterName: ManagedFields:[{Manager:kubectl-create Operation:Update APIVersion:cis.f5.com/v1 TiieldsType:FieldsV1 FieldsV1:{"f:metadata":{"f:labels":{".":{},"f:f5cr":{}}},"f:spec":{".":{},"f:host":{},"f:pools":{},"f:snat":{},"f:virtualServerAddress":{}}}}]} Spec:{Host:*.example.com HostGroup: VirtualServerAddress:10.192.75.118 IPAMLabel: VirtualServerName: VirtuTPSPort:0 Pools:[{Path:/ Service:f5-demo ServicePort:80 NodeMemberLabel: Monitor:{Type:http Send:/ Recv: Interval:20 Timeout:10} Rewrite:}] TLSProfileName: HTTPTraffic: SNAT:auto WAF: RewriteAppRoot: AllowVLANs:[] IRules:[] ServiceIPAddress:[] PolicyName:} Status:{VSAd
```
Will create Jira to prioritize

**Comment by mdditt2000** [comment]
Jira CONTCNTR-3144 for PM

**Comment by mitchellmaler** [comment]
Hello @mdditt2000, do you know the status of this issue? We are still blocked from upgrading the controller.

**Comment by trinaths** [comment]
@mitchellmaler its been taken to 2.9 for review.

**Comment by mitchellmaler** [comment]
@trinaths Here is the line that is causing the issue
https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/pkg/controller/worker.go#L1048

We now have a requirement to encrypt cookies for security reasons which means we need to use the Policy CRD but as we cannot upgrade to 1.7 we are blocked on that too. Do you know a way we can enable an HTTP profile on 1.6 or are we stuck until this bug is resolved?

**Comment by mitchellmaler** [comment]
I believe this is resolved in the later version (tested with 2.9.0) as it doesn't require the host to be set.

**Comment by arzzon** [comment]
Following are my observations with CIS v2.9.0
This works when we don't mention hostname in VS CRs (mention it in TLSProfile) and use TLSProfile with bigIP reference.
Following are all the scenarios those I tried with CIS v2.9.0:
1) If we mention hostname in VS CR then CIS throws the following error:
[ERROR] Same host *.example.com is configured with different VirtualServerAddress :
2) If we don't mention the hostname in VS CR
     2.1) If TLSProfile CR has the reference type as bigIP then it works fine, but any changes made to the TLSProfile CR doesn't trigger an update of the Virtual server. [The reason is that for any update to the TLSProfile the virtual server referencing it must have the same host specified as mentioned in the TLSProfile CR in order for it to be processed again to incorporate these changes]
     2.2) If the reference type is secret then CIS does the certificate verification against the hostname mentioned in the VS CR (here the hostname is empty) and throws the following error:
            Error: Hostname in virtualserver does not match with certificate hostname: x509: certificate is valid for *.example.com
